From: Juergen Gross <jgross@suse.com>
Date: Thu, 2 Mar 2021 08:03:16 +0100
Subject: [PATCH] kABI: Fix kABI after modifying struct __call_single_data
Patch-mainline: Never, kABI fix
References: bsc#1180846

Fix kABI of struct __call_single_data after adding src and dst
elements.

This is trivial, as the struct is 8-byte aligned for 64-bit due to the
pointers in it and there are 4 bytes padding at the end of the struct
where src and dst have been added for 64-bit only.

Signed-off-by: Juergen Gross <jgross@suse.com>
---
 include/linux/smp.h             | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/linux/smp.h b/include/linux/smp.h
index 80d557ef8a11..9f13966d3d92 100644
--- a/include/linux/smp.h
+++ b/include/linux/smp.h
@@ -21,7 +21,9 @@ struct __call_single_data {
 	void *info;
 	unsigned int flags;
 #ifdef CONFIG_64BIT
+#ifndef __GENKSYMS__
 	u16 src, dst;
+#endif
 #endif
 };
 
-- 
2.25.0

